app/models/spree/product_catalog.rb

```diff
@@ -24,7 +26,8 @@ def all_selected?(variants)
 
     def populate_default_items
       variants = Spree::Product.all.map(&:variants_including_master).flatten
-      update_attribute(:item_ids, variants.map(&:id).map(&:to_s))
+      variant_ids = variants.map { |pv| pv.id.to_s }
+      update(item_ids: variant_ids)
```

Comment:
Could you give some thought as to this change to `update`? My understanding was that its main benefit was for return values. Are we expecting the return value of this method to change?

Reply:
The change is twofold:

1. Change the 2-pass maps into one map
2. Don't skip validation (I'm not sure why we were doing so before).

Reply:
I don't recall the exact reasoning for switching from `update_attribute` to `update` but I don't think there is a good reason to skip validations at this point in the code.

Reply:
Oh, that makes sense. I actually didn't realize it was `update_attribute` without the `s`.

Thanks.